omnibus/config/projects/agent.rb

```diff
@@ -197,85 +214,79 @@
+      ## this is a duplicate of the above dependency in linux
+      dependency 'datadog-security-agent-policies'
+    end
   end
-  if ENV['WINDOWS_DDPROCMON_DRIVER'] and not ENV['WINDOWS_DDPROCMON_DRIVER'].empty?
-    dependency 'datadog-windows-procmon-driver'
-    ## this is a duplicate of the above dependency in linux
-    dependency 'datadog-security-agent-policies'
+
+  # this dependency puts few files out of the omnibus install dir and move them
+  # in the final destination. This way such files will be listed in the packages
+  # manifest and owned by the package manager. This is the only point in the build
+  # process where we operate outside the omnibus install dir, thus the need of
+  # the `extra_package_file` directive.
+  # This must be the last dependency in the project.
+  dependency 'datadog-agent-finalize'
+  dependency 'datadog-cf-finalize'
+  if do_package
+    dependency "init-scripts-agent"
   end
 end
-
-# this dependency puts few files out of the omnibus install dir and move them
-# in the final destination. This way such files will be listed in the packages
-# manifest and owned by the package manager. This is the only point in the build
-# process where we operate outside the omnibus install dir, thus the need of
-# the `extra_package_file` directive.
-# This must be the last dependency in the project.
-dependency 'datadog-agent-finalize'
-dependency 'datadog-cf-finalize'
+if do_package
+  dependency "package-artifact"
+  dependency "init-scripts-agent"
+end
 
 if linux_target?
-  extra_package_file '/etc/init/datadog-agent.conf'
-  extra_package_file '/etc/init/datadog-agent-process.conf'
-  extra_package_file '/etc/init/datadog-agent-sysprobe.conf'
-  extra_package_file '/etc/init/datadog-agent-trace.conf'
-  extra_package_file '/etc/init/datadog-agent-security.conf'
-  systemd_directory = "/usr/lib/systemd/system"
-  if debian_target?
-    systemd_directory = "/lib/systemd/system"
-
-    extra_package_file "/etc/init.d/datadog-agent"
-    extra_package_file "/etc/init.d/datadog-agent-process"
-    extra_package_file "/etc/init.d/datadog-agent-trace"
-    extra_package_file "/etc/init.d/datadog-agent-security"
-  end
-  extra_package_file "#{systemd_directory}/datadog-agent.service"
-  extra_package_file "#{systemd_directory}/datadog-agent-process.service"
-  extra_package_file "#{systemd_directory}/datadog-agent-sysprobe.service"
-  extra_package_file "#{systemd_directory}/datadog-agent-trace.service"
-  extra_package_file "#{systemd_directory}/datadog-agent-security.service"
   extra_package_file '/etc/datadog-agent/'
   extra_package_file '/usr/bin/dd-agent'
   extra_package_file '/var/log/datadog/'
 end
 
 # all flavors use the same package scripts
 if linux_target?
+  if do_build && !do_package
+    extra_package_file "#{Omnibus::Config.project_root}/package-scripts/agent-deb"
+    extra_package_file "#{Omnibus::Config.project_root}/package-scripts/agent-rpm"
+  end
   if debian_target?
```

Comment:
```suggestion
  elsif debian_target?
```
I think

Reply:
I enclosed the 2 distribution specific lines in an `if do_package` block, as they don't make sense when building